### DISCUSSION TITLE
Unable to Use cy.select()

### DISCUSSION THREAD

**Discussion by ryan-mee** [cypress-io/cypress#28755] [Questions and Help]
Pretty simple use case, but I just keep getting an error when trying to run cy.select(lng) on a simple language select input.

The command that triggers the error is:

```
cy.get(`[data-testid=select-language-input]`).select('es');
```

It finds the select input element correctly, and then on the select command errors and gives me is:

```
Cypress detected that you returned a promise from a command while also invoking one or more cy commands in that promise.

The command that returned the promise was:

  > cy.select()

The cy command you invoked inside the promise was:

  > cy.wait()

Because Cypress commands are already promise-like, you don't need to wrap them or return your own promise.

Cypress will resolve your command with whatever the final Cypress command yields.

The reason this is an error instead of a warning is because Cypress internally queues commands serially whereas Promises execute as soon as they are invoked. Attempting to reconcile this would prevent Cypress from ever resolving
```

The component being tested is:

```
<select class="languageSwitcher_language-select-input__8tfGT" name="language-select" id="language-select-id" data-testid="select-language-input">
  <option value="en" data-testid="language-option-en">English</option>
  <option value="es" data-testid="language-option-es">Spanish</option>
</select>
```

No idea why its doing this. Has anyone else run into a similar issue?

The project is being run in an NX monorepo and uses the `@badeball/cypress-cucumber-preprocessor` package to allow me to write the tests with Cucumber/Gherkin syntax. Haven't run into this issue with any other commands. I've tried writing it a few different ways but can't get around it.

**Comment by portyu9** [comment]
Hi @ryan-mee 

Your `.select()` syntax and the HTML shown are valid:

```js
cy.get('[data-testid=select-language-input]')
  .select('es')
  .should('have.value', 'es')
```

because your element contains:

```html
<option value="es">Spanish</option>
```

The actual error points somewhere else:

> returned a promise from a command while also invoking one or more cy commands in that promise

and specifically says `cy.wait()` was invoked while `cy.select()` was executing.

The built-in `.select()` does not need a `cy.wait()` to select a normal HTML `<select>`, so I would look in your Cypress support/custom-command code for something that is wrapping or overwriting Cypress commands.

In particular, search for:

```js
Cypress.Commands.overwrite('select', ...)
Cypress.Commands.add('select', ...)
Cypress.Commands.overwrite('click', ...)
```

and for code that mixes:

```js
new Promise(...)
async / await
```

with Cypress commands such as:

```js
cy.wait(...)
cy.wrap(...)
```

Cypress commands already have their own command queue, so this pattern is invalid:

```js
return new Promise((resolve) => {
  cy.wait(1000)
  resolve()
})
```

If you have overwritten a command, the overwrite should normally return the original Cypress chain rather than wrapping Cypress commands in a native Promise:

```js
Cypress.Commands.overwrite(
  'select',
  (originalFn, subject, value, options) => {
    return originalFn(subject, value, options)
  }
)
```

I would also test the simple `.select('es')` assertion in an isolated spec with your custom support registrations temporarily removed. If it then works, add the support/custom-command code back until you identify the overwrite or Promise callback introducing `cy.wait()`.

There isn't enough evidence in the post to blame the Cucumber preprocessor itself.

Please mark this answer as accepted if it helped, thank you.
